testing/tests/test_e2e/test_pebble.py

```diff
@@ -546,3 +546,89 @@ def _on_bar_check_failed(self, event):
     assert foo_infos[0].status == pebble.CheckStatus.DOWN
     assert foo_infos[0].failures == 7
     assert len(bar_infos) == 0
+
+
+def test_pebble_add_layer():
+    class MyCharm(CharmBase):
+        def __init__(self, framework: Framework):
+            super().__init__(framework)
+            framework.observe(self.on.foo_pebble_ready, self._on_foo_ready)
+
+        def _on_foo_ready(self, _):
+            self.unit.get_container("foo").add_layer(
+                "foo",
+                {"checks": {"chk1": {"override": "replace"}}},
+            )
+
+    ctx = Context(MyCharm, meta={"name": "foo", "containers": {"foo": {}}})
+    container = Container("foo", can_connect=True)
+    state_out = ctx.run(
+        ctx.on.pebble_ready(container), state=State(containers={container})
+    )
+    chk1_info = state_out.get_container("foo").get_check_info("chk1")
+    assert chk1_info.status == pebble.CheckStatus.UP
+
+
+def test_pebble_start_check():
```

Comment:
This test could be combined with the test above and parametrised on chk1 plan value.

That's assuming I'm not blind and the rest of the test is actually the same.

Reply:
They are testing distinct behaviours, though, so my preference is to keep them separate. However, they are very similar as you say - I've adjusted them instead so that the start check is more clearly testing two different parts, first that a startup:disabled check doesn't start, and then that it can be started.